### DISCUSSION TITLE
Adding metadata to configs

### DISCUSSION THREAD

**Discussion by justinmoon** [fedimint/fedimint#2719] [General]
The `meta` section of the config is already used for "federation name". It can be useful for other things. For example, if a federation is only intended to run for 1 month the `meta` field could be used to communicate this.

In order for this to work we need:

1) Social consensus on how to interpret these fields. Perhaps something like the BIP process.
2) Ability for guardians to use the Admin UI to propose and accept federation consensus on changes to the `meta` field of the config.

**Comment by dpc** [comment]
> Ability for guardians to use the Admin UI to propose and accept federation consensus on changes to the meta field of the config.

If they are part of the consensus, this falls under the consensus upgrades.

**Reply by elsirion** [reply]
I could imagine making the meta fields part of the DB instead of the config. Since these are simple "propose new value and vote" updates instead of DKG that should be much easier to accomplish.

This doesn't mean it's needed from the beginning, a good MVP would be allowing guardians to specify meta field contents in the setup process. As long as we don't have a way to change them it's probably best to add a layer of indirection by referring to a override file served via HTTPs.

**Reply by otech47** [reply]
so is the best short-term path forward to rely on setting an initial `meta_external_url` during the setup process?

and then longer-term extracting the meta fields from the consensus.json so we don't need to commit to it with DKG?

**Reply by elsirion** [reply]
Yes, for now [`meta_override_url`](https://github.com/fedimint/fedimint/blob/master/docs/meta_fields/meta_override_url.md) is the best solution to the problem.

In the hopefully not-too-distant future the meta fields should become dynamic: https://github.com/fedimint/fedimint/issues/2791
